cpp/src/search/search_ordered.cu

```diff
@@ -68,28 +68,49 @@ std::unique_ptr<column> search_ordered(table_view const& haystack,
 
   auto const comparator = cudf::experimental::row::lexicographic::two_table_comparator(
     matched_haystack, matched_needles, column_order, null_precedence, stream);
-  auto const has_nulls    = has_nested_nulls(matched_haystack) or has_nested_nulls(matched_needles);
-  auto const d_comparator = comparator.less(nullate::DYNAMIC{has_nulls});
+  auto const has_nulls = has_nested_nulls(matched_haystack) or has_nested_nulls(matched_needles);
 
   auto const haystack_it = cudf::experimental::row::lhs_iterator(0);
   auto const needles_it  = cudf::experimental::row::rhs_iterator(0);
 
-  if (find_first) {
-    thrust::lower_bound(rmm::exec_policy(stream),
-                        haystack_it,
-                        haystack_it + haystack.num_rows(),
-                        needles_it,
-                        needles_it + needles.num_rows(),
-                        out_it,
-                        d_comparator);
+  if (cudf::detail::has_nested_columns(haystack) || cudf::get_nested_columns(needles).size() > 0) {
```

Comment:
Why `has_nested_columns` is in the `detail` namespace but `get_nested_columns` is not?

Reply:
Compile error: `error: namespace "cudf" has no member "get_nested_columns"`

Reply:
That's an oversight, `get_nested_columns` has been removed.